apps/server/src/modules/files-storage/service/file-storage.service.spec.ts

```diff
@@ -0,0 +1,188 @@
+import { createMock, DeepMocked } from '@golevelup/ts-jest';
+import { MikroORM } from '@mikro-orm/core';
+import { ObjectId } from '@mikro-orm/mongodb';
+import { Test, TestingModule } from '@nestjs/testing';
+import { EntityId, FileRecord, FileRecordParentType } from '@shared/domain';
+import { FileRecordRepo } from '@shared/repo';
+import { fileRecordFactory, setupEntities } from '@shared/testing';
+import { Logger } from '@src/core/logger';
+import { S3ClientAdapter } from '../client/s3-client.adapter';
+import { FileRecordParams } from '../controller/dto/file-storage.params';
+import { FilesStorageHelper } from '../helper';
+import { FilesStorageService } from './file-storage.service';
+
+describe('FilesStorageService', () => {
+	let module: TestingModule;
+	let service: FilesStorageService;
+	let fileRecordRepo: DeepMocked<FileRecordRepo>;
+	let storageClient: DeepMocked<S3ClientAdapter>;
+	let filesStorageHelper: DeepMocked<FilesStorageHelper>;
+	let orm: MikroORM;
+	let fileRecord: FileRecord;
+	let fileRecords: FileRecord[];
+
+	const entityId: EntityId = new ObjectId().toHexString();
+	const userId: EntityId = new ObjectId().toHexString();
+	const schoolId: EntityId = new ObjectId().toHexString();
+
+	beforeAll(async () => {
+		orm = await setupEntities();
+
+		fileRecord = fileRecordFactory.buildWithId({ name: 'text.txt' });
+	});
+
+	afterAll(async () => {
+		await orm.close();
+	});
+
+	beforeEach(async () => {
+		module = await Test.createTestingModule({
```

Comment:
Do we need the same modul all the time? Maybe then we can use before instead of beforeEach.
We have no needed cleanup of DBs or of the unit of work.